esmf_regrid/tests/unit/experimental/unstructured_scheme/test_GridToMeshESMFRegridder.py

```diff
@@ -0,0 +1,135 @@
+    tgt.data[:] = 0
+
+    def _add_metadata(cube):
+        result = cube.copy()
+        result.units = "K"
+        result.attributes = {"a": 1}
+        result.standard_name = "air_temperature"
+        scalar_height = AuxCoord([5], units="m", standard_name="height")
+        scalar_time = DimCoord([10], units="s", standard_name="time")
+        result.add_aux_coord(scalar_height)
+        result.add_aux_coord(scalar_time)
+        return result
+
+    src = _add_metadata(src)
+    src.data[:] = 1  # Ensure all data in the source is one.
+    regridder = GridToMeshESMFRegridder(src, tgt)
+    result = regridder(src)
+    src_T = src.copy()
+    src_T.transpose()
+    result_transposed = regridder(src_T)
+
+    expected_data = np.ones([n_lats, n_lons])
+    expected_cube = _add_metadata(tgt)
+
+    # Lenient check for data.
+    assert np.allclose(expected_data, result.data)
+    assert np.allclose(expected_data, result_transposed.data)
+
+    # Check metadata and scalar coords.
+    expected_cube.data = result.data
+    assert expected_cube == result
+    expected_cube.data = result_transposed.data
+    assert expected_cube == result_transposed
+
+
+def test_multidim_cubes():
+    """
+    Test for :func:`esmf_regrid.experimental.unstructured_scheme.regrid_rectilinear_to_unstructured`.
+
+    Tests with multidimensional cubes. The source cube contains
+    coordinates on the dimensions before and after the grid dimensions.
+    """
+    tgt = _flat_mesh_cube()
+    mesh = tgt.mesh
+    mesh_length = mesh.connectivity(contains_face=True).shape[0]
+    n_lons = 6
+    n_lats = 5
+    lon_bounds = (-180, 180)
+    lat_bounds = (-90, 90)
+    grid = _grid_cube(n_lons, n_lats, lon_bounds, lat_bounds, circular=True)
+
+    h = 2
+    t = 3
+    height = DimCoord(np.arange(h), standard_name="height")
+    time = DimCoord(np.arange(t), standard_name="time")
+
+    src_data = np.empty([t, n_lats, n_lons, h])
+    src_data[:] = np.arange(t * h).reshape([t, h])[:, np.newaxis, np.newaxis, :]
+    cube = Cube(src_data)
+    cube.add_dim_coord(grid.coord("latitude"), 1)
+    cube.add_dim_coord(grid.coord("longitude"), 2)
+    cube.add_dim_coord(time, 0)
+    cube.add_dim_coord(height, 3)
+
+    regridder = GridToMeshESMFRegridder(grid, tgt)
+    result = regridder(cube)
+
+    # Lenient check for data.
+    expected_data = np.empty([t, mesh_length, h])
+    expected_data[:] = np.arange(t * h).reshape(t, h)[:, np.newaxis, :]
+    assert np.allclose(expected_data, result.data)
+
+    mesh_coord_x, mesh_coord_y = mesh.to_MeshCoords("face")
+    expected_cube = Cube(expected_data)
+    expected_cube.add_dim_coord(time, 0)
+    expected_cube.add_aux_coord(mesh_coord_x, 1)
+    expected_cube.add_aux_coord(mesh_coord_y, 1)
+    expected_cube.add_dim_coord(height, 2)
+
+    # Check metadata and scalar coords.
+    result.data = expected_data
+    assert expected_cube == result
+
+
+def test_invalid_mdtol():
```

Comment:
Do you have a test to show that mdtol works?  i.e. if I set mdtol=0 and that can't be satisfied, does it error?

Reply:
this is probably worth adding, yes. this shouldn't result in an error but in masked data, but it is still probably worth testing here.